client/blocks/eligibility-warnings/index.jsx

```diff
@@ -2,144 +2,100 @@
-// TODO: update supportUrls and maybe create similar mapping for warnings
-function getHoldMessages( translate ) {
-	return {
-		PLACEHOLDER: {
-			title: '',
-			description: '',
-			supportUrl: '',
-		},
-		NO_BUSINESS_PLAN: {
-			title: translate( 'Business plan required' ),
-			description: translate( 'This feature is only allowed on sites with a business plan.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		NO_VIP_SITES: {
-			title: translate( 'VIP site' ),
-			description: translate( 'This feature is not supported on VIP sites.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		NO_WPCOM_NAMESERVERS: {
-			title: translate( 'No WordPress.com name servers' ),
-			description: translate( 'Your custom domain must point to WordPress.com name servers.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		NON_ADMIN_USER: {
-			title: translate( 'Admin access required' ),
-			description: translate( 'Only site administrators are allowed to use this feature.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		NOT_USING_CUSTOM_DOMAIN: {
-			title: translate( 'No custom domain' ),
-			description: translate( 'Your site must use a custom domain to use this feature.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		NOT_DOMAIN_OWNER: {
-			title: translate( 'Not a custom domain owner' ),
-			description: translate( 'You must be the owner of the primary domain subscription to use this feature.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		SITE_GRAYLISTED: {
-			title: translate( 'Flagged site' ),
-			description: translate( 'This feature is not supported on sites that are not in good standing.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		},
-		SITE_PRIVATE: {
-			title: translate( 'Private site' ),
-			description: translate( 'This feature is not supported on private sites.' ),
-			supportUrl: 'https://support.wordpress.com/'
-		}
-	};
-}
-
-const EligibilityWarnings = props => {
-	const {
-		translate,
-		backUrl,
-		onProceed,
-		siteId,
-		isEligible,
-		eligibilityData,
-		isPlaceholder,
-	} = props;
+export const EligibilityWarnings = ( {
+	backUrl,
+	eligibilityData,
+	hasBusinessPlan,
+	isEligible,
+	isJetpack,
+	isPlaceholder,
+	onProceed,
+	siteId,
+	siteSlug,
+	translate,
+} ) => {
+	const context = -1 !== backUrl.indexOf( 'plugins' ) ? 'plugins' : 'themes';
 
-	const holdMessages = getHoldMessages( translate );
-	const holds = get( eligibilityData, 'eligibilityHolds', [ 'PLACEHOLDER', 'PLACEHOLDER' ] );
 	const warnings = get( eligibilityData, 'eligibilityWarnings', [] );
 
-	const classes = classNames( {
-		'eligibility-warnings__message': true,
-		'eligibility-warnings__placeholder': isPlaceholder,
-	} );
+	let holds = get( eligibilityData, 'eligibilityHolds', [ 'PLACEHOLDER', 'PLACEHOLDER' ] );
+	const bannerHolds = filter( holds, hold => -1 !== [ 'NO_BUSINESS_PLAN', 'NOT_USING_CUSTOM_DOMAIN' ].indexOf( hold ) );
+	holds = difference( holds, bannerHolds );
```

Comment:
do we have a reason to mutate this so quickly instead of just declaring it? it's an odd use of `filter` and `difference` - did we mean to use a different set function? what's the goal here? to remove the banner holds from the list of holds?